src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp

```diff
@@ -31,11 +32,57 @@ namespace CDASimAdapter{
         PLOG(logINFO) << "CARMA-Simulation connection is successful!" << std::endl;
         return _connected;
     }
-    bool CDASimConnection::carma_simulation_handshake(const std::string &simulation_ip, const uint simulation_registration_port, 
+
+    std::string CDASimConnection::get_handshake_json(const uint infrastructure_id, const std::string &local_ip,  const uint time_sync_port, const uint v2x_port, 
+                                const WGS84Point &location)
+
+    {
+        Json::Value message;   
+        std::string message_str = "";
+        try
+        {
+            message["rxMessageIpAddress"] = local_ip;
+            message["infrastructureId"] = infrastructure_id;
+            message["rxMessagePort"] = v2x_port;
+            message["timeSyncPort"] = time_sync_port;
+            message["location"]["latitude"] = location.Latitude;
+            message["location"]["longitude"] = location.Longitude;
+            message["location"]["elevation"] = location.Elevation;
+            Json::StyledWriter writer;
+            message_str = writer.write(message);
+    
+        }
+        catch(std::exception e)
+        {
+            PLOG(logERROR) << "Encountered runtime error when creating JSON message: " << e.what() << std::endl;
+            return "";
+        }
+        return message_str;
+    }
+
+    bool CDASimConnection::carma_simulation_handshake(const std::string &simulation_ip, const uint infrastructure_id, const uint simulation_registration_port, 
                                 const std::string &local_ip,  const uint time_sync_port, const uint v2x_port, 
-                                const WGS84Point &location) {
-        // TODO: Standup UDP Server and Client for registration and implement handshake
-        return true;                                
+                                const WGS84Point &location) 
+    {
+        // Create JSON message with the content 
+        std::string payload = get_handshake_json(infrastructure_id, local_ip, time_sync_port, v2x_port, location);
+        if (payload.compare("") == 0)
```

Comment:
String has an isEmpty method you should use instead of this.

Reply:
added